en/docs/develop/customizations/advanced-ui-customization.md

```diff
@@ -0,0 +1,402 @@
+
+**source**
+
+This directory contains the Javascript source codes. It includes both the implementation and tests. Note that these files do not get loaded into the browser at runtime. 
+We have split the implementation section into two aspects; data and components. Data contains pure javascript implementations of data handling (REST API innovation), utility functions, and data models (user class), whereas the components directory includes the ReactJS component implementations.  The complete component hierarchy of the publisher portal is in this folder. This separation makes the data handling elements shareable and usable in non- ReactJS based projec
+
+**jaggery.conf**
+
+This is a configuration file used for web apps in Jaggery web server. It contains
+
+routing rules(urlMappings)
+security constraints
+welcomeFiles
+and other jaggery app-related configurations.
+
+If you are new to JaggeryJS, JaggeryJS is a javascript backend server that can render jaggery style .jag (like Pug) templates into HTML. Simply put, it’s similar to NodeJS + Express.
+
+**Others**
+
+This folder includes the rest of the files which are runtime configurations for eslint, jest, webpack, and npm. 
+
+## Making advanced UI customizations to the WSO2 API-M Publisher and Developer Portal UIs
+
+Follow the instructions below to add advanced UI customizations to the Developer Portal and/or Publisher. 
+
+!!! note "Before you Begin..."  
+    Make sure the following is installed in your system.
+        - **NodeJS** (minimum 8.12.0) - This is a platform required for ReactJS development.
+        - **NPM**(minimum 5.7.0)
+
+1. Navigate to the `<API-M_HOME>/repository/deployment/server/jaggeryapps/<APP-ROOT>` directory in a terminal and run the following command. ( `<APP-ROOT>` is publisher or devportal ). This will install the local package dependencies in the publisher/devportal applications.
+
+     ```js
+     npm ci
+     ```
+
+     
+2. Build the project (enabling hot deployment) 
+
+     Run the following command to start the npm build. Note that it will continuously watch for any changes and rebuild the project. For more information see [Building the source for hot deployment](#Building the source for hot deployment) 
+
+    ```
+    npm run build:dev
+    ```
+    !!! note "Production deployment"
+        The development build is not optimized and contains a large bundle size. Make sure to use the production build when the customizations are ready for production. Use the following command to get the production-ready build.
+        
+        ```
+        npm run build:prod
+        ```
+3. Customizing the UI 
+
+     - If you want to override a specific React component or a file from the `<WEBAPP>/source/src/` directory, you need to make the changes in the following directory by only copying the desired file(s). For more information see [Overriding the UI components](#Overriding the UI components)
```

Comment:
add the correct link here

Reply:
This section modified and the link is removed